### PR TITLE
[ci][auto/3] add ci state machine bot to continuous run

### PR DESCRIPTION
**Pull request by can-anyscale**
Add CI state machine bot to continuous run.

Test:
- CI
- Ran the bot locally

### CODE REVIEW

File: .buildkite/others.rayci.yml [hunk 1]
```diff
@@ -79,3 +79,12 @@ steps:
         --test-env=USERPROFILE
     depends_on: windowsbuild
     soft_fail: true
+
+  # bot
+  - label: ":robot_face: CI state machine bot"
+    tags: skip-on-premerge
+    if: build.branch == "master" && build.env("RAYCI_CONTINUOUS_BUILD") == "1"
```
Comment (aslonnie) [inline .buildkite/others.rayci.yml:86]: why is `RAYCI_CONTINUOUS_BUILD` set to 1? on nighly?

**Review by aslonnie — APPROVED** [review]

File: .buildkite/others.rayci.yml (hunk 1 above)
Comment (can-anyscale) [reply .buildkite/others.rayci.yml:86]: I'll create a schedule run on postmerge with that env set: https://buildkite.com/ray-project/postmerge/settings/schedules. It's not there yet.

**Review by aslonnie — APPROVED** [review]

File: .buildkite/others.rayci.yml [hunk 2]
```diff
@@ -79,3 +79,12 @@ steps:
...
+    instance_type: small
+    commands:
+      - bazel run //ci/ray_ci/automation:state_machine_bot -- --production
+    soft_fail: true
```
Comment (aslonnie) [inline .buildkite/others.rayci.yml:90]: why soft fail?

Comment (can-anyscale) [reply .buildkite/others.rayci.yml:90]: yes it should always pass, let remove it

**Pull request merged**